### PR TITLE
Faster Docker Builds with cargo-chef

### PR DESCRIPTION
**Pull request by mnaamani**
Fixes
https://github.com/Joystream/joystream/issues/3036
https://github.com/Joystream/joystream/issues/2853

Note: For the joystream-node.Dockerfile I'm no longer copying in the entire git repo (including .git/) so during the build step the custom build script cannot get the git commit hash to append to the node version string output of `--version`. So we get "unknown" instead of commit hash:

```
docker run joystream/node:test --version
node 7.3.0-unknown
```

We can set env variable manually: `SUBSTRATE_CLI_GIT_COMMIT_HASH` if we wish, but wondering if it makes any sense to add it at all.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1203052029537863) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by mnaamani** [comment]
I realized I made some changes the `joystream-node-docker-dev.yml` workflow which should have been done for the `run-network-tests.yml` workflow instead. I'll revert and update the latter.

File: joystream-node.Dockerfile [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
+COPY runtime-modules ./runtime-modules
+RUN cargo chef prepare --recipe-path /joystream/recipe.json
+
+FROM rust AS cacher
+LABEL description="Cargo chef cook dependencies"
+WORKDIR /joystream
+COPY --from=planner /joystream/recipe.json /joystream/recipe.json
+ENV WASM_BUILD_TOOLCHAIN=nightly-2022-05-11
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --recipe-path /joystream/recipe.json
```
Comment (dobertRowneySr) [inline joystream-node.Dockerfile:28]: Is the `recipe.json` set up manually, if so should we vc it?

**Review by dobertRowneySr — APPROVED** [review]
LGTM, I just left a small question

File: joystream-node.Dockerfile (hunk 1 above)
Comment (mnaamani) [reply joystream-node.Dockerfile:28]: the recipie.json file is produced in the previous docker layer "planner" with the command `cargo chef prepare ..` It is copied into this layer with the `COPY` command on line 25.

**Pull request merged**